spec/rubocop/cop/rspec/sort_metadata_spec.rb

```diff
@@ -23,6 +23,50 @@
     RUBY
   end
 
+  it 'registers an offense when a symbol metadata is before second docstring ' \
+     'argument' do
+    expect_offense(<<~RUBY)
+      RSpec.describe 'Something', :a, :z, 'second docstring' do
+                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^ Sort metadata alphabetically.
+      end
+    RUBY
+
+    expect_correction(<<~RUBY)
+      RSpec.describe 'Something', 'second docstring', :a, :z do
+      end
+    RUBY
+  end
+
+  it 'registers an offense when a symbol metadata is before a variable ' \
+     'argument' do
+    expect_offense(<<~RUBY)
+      RSpec.describe 'Something', :a, :z, variable, foo: :bar do
+                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^ Sort metadata alphabetically.
+      end
+    RUBY
+
+    expect_correction(<<~RUBY)
+      RSpec.describe 'Something', variable, :a, :z, foo: :bar do
+      end
+    RUBY
+  end
+
+  it 'does not register an offense when a symbol metadata is before a ' \
+     'variable argument being the last argument as it could be a hash' do
+    expect_no_offenses(<<~RUBY)
+      RSpec.describe 'Something', :a, :z, some_hash do
```

Comment:
What if it's a
```
RSpec.describe 'Something', :z, :a, some_hash do
```
?

Reply:
Then it would register an offense and would autocorrect it like this:

```ruby
RSpec.describe 'Something', :a, :z, some_hash do
```

This test was inspired from this `MetadataStyle` cop test:

https://github.com/rubocop/rubocop-rspec/blob/74d78371f4495099cf23bcf93b8af7ef711aec26/spec/rubocop/cop/rspec/metadata_style_spec.rb#L354-L364

It ensures that a variable is not moved at the start of the arguments list if this variable is at the end of the arguments list, because it could hold a hash.

Should I update the spec to swap `:a` and `:z` and expect the offense and correction as described above?